### PR TITLE
Ingester shutdown marker improvements

### PR DESCRIPTION
**Pull request by 56quarters**
#### What this PR does

This PR improves the way shutdown markers for ingesters work. Namely it:

* Removes the shutdown marker while shutting down. The volumes used by ingesters are reused if an ingester is stopped (scaled down) but then later started again with the same name. Removing the marker during shutdown prevents the ingester from unregistering and flushing the next time it is stopped after having already performed its work for "permanently" shutting down.
* Prevents changes to the shutdown marker while the ingester is not running. This prevents the rollout-operator from calling the `prepare-shutdown` endpoint again while the ingester is already in the process of shutting down and about to remove the shutdown marker as noted previously.

#### Which issue(s) this PR fixes or relates to

N/A

#### Checklist

- [X] Tests updated
- [ ] Documentation added
- [ ] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

**Review by pracucci — COMMENTED** [review]
Can you fix it for store-gateway too, please?

**Review by pstibrany — APPROVED** [review]
Looks good, thanks.

**Review by dimitarvdimitrov — APPROVED** [review]
LGTM

File: pkg/ingester/ingester.go [hunk 1]
```diff
@@ -2478,9 +2490,9 @@ func (i *Ingester) PrepareShutdownHandler(w http.ResponseWriter, r *http.Request
 		}
 
 		if exists {
-			util.WriteTextResponse(w, "set")
+			util.WriteTextResponse(w, "set\n")
```
Comment (pracucci) [inline pkg/ingester/ingester.go:2493]: Shouldn't this change done for the store-gateway too?

File: pkg/ingester/ingester.go [hunk 2]
```diff
@@ -448,6 +448,12 @@ func (i *Ingester) stopping(_ error) error {
 		level.Warn(i.logger).Log("msg", "failed to stop ingester lifecycler", "err", err)
 	}
 
+	// Remove the shutdown marker if it exists since we are shutting down
+	shutdownMarkerPath := shutdownmarker.GetPath(i.cfg.BlocksStorageConfig.TSDB.Dir)
```
Comment (pracucci) [inline pkg/ingester/ingester.go:452]: Shouldn't this change done for the store-gateway too?

**Pull request merged**